src/main/kotlin/cacophony/grammar/syntaxtree/Expression.kt

```diff
@@ -0,0 +1,229 @@
+package cacophony.grammar.syntaxtree
+
+import cacophony.utils.Location
+
+sealed class Expression(val range: Pair<Location, Location>) { // everything in cacophony is an expression
+    class Variable(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    // should Type also inherit expression? if so, can we do for example "; Bool;" in our language?
+    class Type(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class VariableDeclaration(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class VariableTypedDeclaration(
+        range: Pair<Location, Location>,
+        variable: VariableDeclaration,
+        type: Type,
+    ) : Expression(range)
+
+    class Literal(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class Keyword(
+        range: Pair<Location, Location>,
+    ) : Expression(range) {
+        class KeywordReturn(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordBreak(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+    }
+
+    // expression in parenthesis
+    class NestedExpression(
+        range: Pair<Location, Location>,
+        val expression: Expression,
+    ) : Expression(range)
+
+    // series of expressions separated by semicolons, returning value of the last one
+    // we agreed that (x; y;) returns Unit - should we create artificial class for empty expression after the last semicolon,
+    // or will we handle it in a different way?
+    class SubsequentExpressions(
+        range: Pair<Location, Location>,
+        vararg val expressions: Expression,
+    ) : Expression(range)
```

Comment:
i wonder if we could merge it into one, as currently `NestedExpression` is just a special case of `SubsequentExpressions` for one expression

Reply:
I will think about it for a moment if it wont mess anything with priority, also if it will be clear for later work

Reply:
semantically they are the same, but I'm not sure if merging will be convenient for further implementation, so I will leave them separate for now